aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/VpcLatticeV2RequestEvent.java

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+package com.amazonaws.services.lambda.runtime.events;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.annotation.Nullable;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <a href="https://docs.aws.amazon.com/vpc-lattice/latest/ug/lambda-functions.html">Lambda functions as targets in VPC Lattice</a>
+ *
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(setterPrefix = "with")
+public class VpcLatticeV2RequestEvent {
+
+    private String version;
+    private String path;
+    private String method;
+    private Map<String, List<String>> headers;
+    @Nullable
+    private Map<String, String> queryStringParameters;
+    private RequestContext requestContext;
+    private String body;
+    @Nullable
+    private Boolean isBase64Encoded;
+
+    @Data
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @Builder(setterPrefix = "with")
+    public static class RequestContext {
+        private String serviceNetworkArn;
+        private String serviceArn;
+        private String targetGroupArn;
+        private Identity identity;
+        private String region;
+        /**
+         * Number of microseconds from the epoch
+         */
+        private String timeEpoch;
+
+        public LocalDateTime getlocalDateTime() {
+            // Truncating to milliseconds. Java 8 only supports Epoch with precision of milliseconds. Microseconds are supported in Java 9
+            long epochMilli = Long.parseLong(timeEpoch.substring(0, timeEpoch.length() - 3));
```

Comment:
I'd add a null check even if the field is supposed to be always present. In addition provide a better method name for `getlocalDateTime`.

Reply:
if it is null, we'll get NPE in this line - and fail fast. I think this is expected behaviour in that case, as there's no reasonable fallback.

Reply:
On another note, I'd avoid truncating string in this line
my suggestion would be
```
long epochMicroseconds = Long.parseLong(timeEpoch);
long epochMilliseconds = epochMicroseconds / 1000;

return Instant.ofEpochMilli(epochMilliseconds).atZone(ZoneOffset.UTC).toLocalDateTime();
```

Reply:
Done